lib/network_interface/worker.ex

```diff
@@ -13,45 +13,105 @@
 # limitations under the License.
 
 defmodule Nerves.NetworkInterface.Worker do
+  @moduledoc """
+  Working for NetworkInterface. 
+  See `Nerves.NetworkInterface` for more details.
+  """
+
   use GenServer
   require Logger
 
-  @moduledoc false
+  defstruct port: nil
+
+  @typedoc "State of the GenServer"
+  @type state :: %__MODULE__{port: port}
```

Comment:
I have a few projects with structs that have `state` too. I like the same idea that you have. What I've started doing in many cases is moving it to its own module. `Nerves.NetworkInterface.Worker.State.t` although I often give it a name that is more in line with what the data holds than `State`. I think that goes back to my `OO` days also. The nice thing that I have found with that is that functions that interact with the data live in a module who has the entire purpose of poking and transforming that data. Also, as the state becomes more complex I can end up with a `State` struct that might hold other structs like `Port` or `Config`. `Config` seems to be one I keep ending at more and more. I put many interactors into the `Config`. I guess it is like passing a group of options to a command line process.

Now that I'm looking at the possibility of `State` ending up becoming complex I think that is where I would go. I'm not talking about changing this. I'm just thinking out loud. That is how I work. Sorry if I'm rambling.

This is awesome work @ConnorRigby. Thanks for doing this.